packages/measure/src/measure-renders.tsx

```diff
@@ -69,14 +81,19 @@ export async function measureRender(ui: React.ReactElement, options?: MeasureOpt
     runResults.push({ duration, count });
   }
 
+  if (hasUnnecessaryRender) {
+    const testName = expect.getState().currentTestName;
+    logger.warn(`test "${testName}" has unnecessary renders. Please update your code to avoid unnecessary renders.`);
+  }
+
   if (hasTooLateRender) {
     const testName = expect.getState().currentTestName;
     logger.warn(
       `test "${testName}" still re-renders after test scenario finished.\n\nPlease update your code to wait for all renders to finish.`
     );
   }
 
-  return processRunResults(runResults, warmupRuns);
+  return processRunResults(runResults, warmupRuns, components);
```

Comment:
instead of passing `compoennt` for analysis to `processRunResults` which is also used by `measureFunction` lets call function to detect redundant renders before calling it.